Using fire extinguisher in help intent is helpful

coiax opened the pull request:
:cl: coiax
rscadd: Using the fire extinguisher on yourself or others while in HELP intent will spray you or them, rather than beat their probably flaming body with a metal object.
tweak: Alt-clicking extinguishers to empty them now wets the floor underneath them.
/:cl:

coiax committed:
Make the floor wet when you empty the extinguisher

coiax committed:
Added help to extinguisher/attack but bugged

It's not triggering the afterattack proc to spray the water...

Supermichael777 commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -160,5 +160,7 @@
 /obj/item/weapon/extinguisher/proc/EmptyExtinguisher(var/mob/user)
 	if(loc == user && reagents.total_volume)
 		reagents.clear_reagents()
+		loc.MakeSlippery(TURF_WET_WATER)
```
I'm sure theirs a proper way to apply reagents to turf and that this isn't it.

Supermichael777 commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -46,6 +46,17 @@
+		// If we're in help intent, don't bash anyone with the
+		// extinguisher
+		user.visible_message("[user] targets [M] with \the [src]", "<span class='info'>You target [M] with \the [src].</span>")
+		// FIXME the fire extinguisher isn't triggering, don't know
```
call afterattack because your returning before calling the parent.
afterattack(m, user) should do it

coiax replied:
no, afterattack is called by the caller of attack.

coiax committed:
Removed check preventing self target extinguisher

The comment on this code segment implies it's to stop the fire
extinguisher from going off when put in a bag. Except it doesn't go
off when put in a bag, even without this code check.

As such, it was stopping self targeting with a fire extinguisher.

Make sure the safety is off though!

PKPenguin321 commented:
I kinda want to -1 this on the basis that this has been one of the staple new player jokes for ages (along with HALP HOW 2 SWITCH HANDS)

coiax commented:
Yes, but isn't this what Help intent is for?

Zilenan91 commented:
Nah let's keep this it's hilarious

PKPenguin321 commented:
![-1](http://i.imgur.com/z6w988l.jpg)

KazeEspada commented:
Except that people being ignitable is only 2-3 years old newfags.

PKPenguin321 commented:
doesn't change the fact that it's funny

Shadowlight213 commented on code/game/objects/items/weapons/extinguisher.dm:
```diff
@@ -160,5 +167,11 @@
+		var/turf/T = get_turf(loc)
+		if(istype(T, /turf/open))
+			var/turf/open/theturf = T
+			theturf.MakeSlippery(TURF_WET_WATER)
```
Stealth change. This should probably be in the changelog

coiax replied:
Noted. Tbh though, I didn't even know you could alt click extinguishers to empty them.

PKPenguin321 commented:
rip
